### PR TITLE
refactor: switch intraledger-send use-case to use new PaymentFlowBuilder

### PR DESCRIPTION
**Pull request by vindard**
## Description

This PR adds a refactored version of the send-intraledger use-case methods that use the `PaymentFlowBuilder` instead to correctly handle all currency conversions.

### CODE REVIEW

File: test/integration/02-user-wallet/02-send-lightning.spec.ts [hunk 1]
```diff
@@ -1509,6 +1510,161 @@ describe("USD Wallets - Lightning Pay", () => {
...
+        senderWalletId: walletIdUsdB,
+        senderAccount: accountB,
+        recipientWalletId: walletIdUsdA,
+        senderAmountInvoice: usdSendAmount,
+        recipientAmountInvoice: usdSendAmount,
+      })
+      expect(res).not.toBeInstanceOf(Error)
+    })
+
+    it.skip("sends 10 sats (e.g. reward) from BTC wallet to USD wallet", async () => {
```
[the remark this answers is not in the document]
Comment (UncleSamtoshi) [reply test/integration/02-user-wallet/02-send-lightning.spec.ts:1651]: What is the issue?

File: src/app/payments/helpers.ts [hunk 2]
```diff
@@ -145,20 +152,27 @@ export const constructPaymentFlowBuilder = async ({
     usdFromBtcMidPriceFn,
     btcFromUsdMidPriceFn,
   })
-  const builderWithInvoice = uncheckedAmount
-    ? paymentBuilder.withNoAmountInvoice({ invoice, uncheckedAmount })
-    : paymentBuilder.withInvoice(invoice)
+  const builderWithInvoice = invoice
```
Comment (bodymindarts) [inline src/app/payments/helpers.ts:155]: Not so happy about this 3 way logic in a helpers method. Its not unit tested.
Either we should unify the builder interface and push the logic into the builder or have different entry points ie un-merge the helper function.

File: src/app/payments/helpers.ts [hunk 3]
```diff
@@ -145,20 +152,27 @@ export const constructPaymentFlowBuilder = async ({
...
   let builderAfterRecipientStep:
     | LPFBWithRecipientWallet<WalletCurrency, WalletCurrency>
     | LPFBWithError
   if (builderWithSenderWallet.isIntraLedger()) {
-    const recipientDetails = await recipientDetailsFromInvoice(invoice)
-    if (recipientDetails instanceof Error) return recipientDetails
-    builderAfterRecipientStep =
-      builderWithSenderWallet.withRecipientWallet(recipientDetails)
+    const recipientDetailsForBuilder =
+      invoice === undefined
```
Comment (bodymindarts) [inline src/app/payments/helpers.ts:168]: Again don't like this logic - I think the intra-ledger and non-intra-ledger use case should perhaps not be merged as they can be clearly distinguished higher in the call stack.

File: src/app/payments/send-intraledger.ts [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+  const recipientWallet = await WalletsRepository().findById(recipientWalletId)
+  if (recipientWallet instanceof Error) return recipientWallet
+  const { accountId: recipientAccountId, currency: recipientWalletCurrency } =
+    recipientWallet
+
+  const recipientAccount = await AccountsRepository().findById(recipientAccountId)
+  if (recipientAccount instanceof Error) return recipientAccount
+  const { username: recipientUsername } = recipientAccount
+
+  const builderWithConversion = await constructPaymentFlowBuilder({
```
Comment (bodymindarts) [inline src/app/payments/send-intraledger.ts:101]: This is no longer part of `validate...`.

File: src/app/payments/send-intraledger.ts [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+    },
+    usdFromBtc: dealer.getCentsFromSatsForImmediateBuy,
+    btcFromUsd: dealer.getSatsFromCentsForImmediateSell,
+  })
+  if (builderWithConversion instanceof Error) return builderWithConversion
+
+  const paymentFlow = await builderWithConversion.withoutRoute()
+  if (paymentFlow instanceof Error) return paymentFlow
+
+  if (paymentFlow instanceof Error) return paymentFlow
```
Comment (bodymindarts) [inline src/app/payments/send-intraledger.ts:119]: Duplication of line 117

File: src/app/payments/send-intraledger.ts [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+  return LockService().lockWalletId(
+    { walletId: senderWallet.id, logger },
+    async (lock) => {
+      const balance = await LedgerService().getWalletBalanceAmount(senderWallet)
+      if (balance instanceof Error) return balance
+
+      const balanceCheck = paymentFlow.checkBalanceForSend(balance)
+      if (balanceCheck instanceof Error) return balanceCheck
+
+      const priceRatio = PriceRatio({
```
Comment (bodymindarts) [inline src/app/payments/send-intraledger.ts:187]: Duplicate of line 155?

**Review by bodymindarts — COMMENTED** [review]
Where are the unit tests to document the additional `PaymentFlowBuilder` methods? (ie `withoutInvoice`?

File: test/integration/02-user-wallet/02-send-lightning.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vindard) [reply test/integration/02-user-wallet/02-send-lightning.spec.ts]: Update: `addEarn` was switched to only pay out to BTC wallets; and amounts less than 1 cent in value for intraledger payments now fail by default (with new tests added)

**Review by UncleSamtoshi — APPROVED** [review]

**Pull request merged**